app/products/calls/state/actions.ts

```diff
@@ -217,6 +218,10 @@ export const newCurrentCall = (serverUrl: string, channelId: string, myUserId: s
 
 export const myselfLeftCall = () => {
     setCurrentCall(null);
+
+    // Remove the call screen, and in some situations it needs to be removed twice before actually being removed.
+    Navigation.pop(Screens.CALL).catch(() => null);
+    Navigation.pop(Screens.CALL).catch(() => null);
```

Comment:
I certainly don't want to delay the PR further but I just can't help myself. Why would we have to call the same thing twice in a row? Are these synchronous calls?

Reply:
The first call is all we need most of the time. But if the user is viewing the call thread from within the call screen, in that case we have to call it twice. If it's not on the stack for the second call, there's no harm. Hope that helps!

Reply:
Okay so in such case we are essentially stacking it twice as well at some point throughout the call.

Reply:
@streamer45 Nope, not stacking twice -- it looks like what's happening is: if the thread view is the direct child of the call screen, popping the call screen actually pops the thread screen (bringing us to the call screen). The second pop pops the call screen.

Reply:
Ah interesting there's some more advanced relation going on then. Not very intuitive but I am not going to object :p